From a4d323d8a72cb7cf8907d87a070a9ad5d8e9c677 Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@jolla.com>
Date: Thu, 21 Aug 2014 12:44:30 +0300
Subject: [PATCH 1/2] Tweak UA for Facebook and Engadget to get images of
 proper sizes

This user agent string (or something like this) could also
be used as a testing platform for new default user agent.
---
 jscomps/UserAgentOverrideHelper.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/jscomps/UserAgentOverrideHelper.js b/jscomps/UserAgentOverrideHelper.js
index 5275eaf..307ff01 100644
--- a/jscomps/UserAgentOverrideHelper.js
+++ b/jscomps/UserAgentOverrideHelper.js
@@ -48,6 +48,8 @@ var UserAgent = {
   GOOGLE_MAPS_DOMAIN: /(^|\.)maps\.google\.com$/,
   YOUTUBE_DOMAIN: /(^|\.)youtube\.com$/,
   NOKIA_HERE_DOMAIN: /(^|\.)here\.com$/,
+  FACEBOOK_DOMAIN: /(^|\.)facebook\.com$/,
+  ENGADGET_DOMAIN: /(^|\.)engadget.com$/,
   _customUA: null,
 
   getCustomUserAgent: function() {
@@ -105,6 +107,9 @@ var UserAgent = {
         if (!ua.contains("Mobile")) {
           return ua.replace("X11", "Android").replace("Unix", "Android").replace("Linux", "Mobile");
         }
+      } else if (this.FACEBOOK_DOMAIN.test(aUri.host) || this.ENGADGET_DOMAIN.test(aUri.host)) {
+        // Facebook works best when UA is a copy of Firefox for Android, but without "Android" keyword
+        return "Mozilla/5.0 (Sailfish; Mobile; rv:31.0) Gecko/31.0 Firefox/31.0";
       }
     }
 
-- 
2.1.4

